draft-ietf-oauth-sd-jwt-vc.md

```diff
@@ -478,30 +481,26 @@ MUST NOT be used.
 
 # Type Metadata {#type-metadata}
 
-A type is associated with metadata defining, for example, information about the type, a schema defining which claims MAY or MUST appear in the SD-JWT VC, and how they can be displayed.
+A SD-JWT VC type, i.e., the `vct` value, is associated with Type Metadata defining, for example, information about the type or a schema defining which claims MAY or MUST appear in the SD-JWT VC.
 
-This section defines Type Metadata that can be associated with a type of a SD-JWT VC as well as a method for retrieving the Type Metadata and processing rules. This Type Metadata is intended to be used, among other things, for the following purposes:
+This section defines Type Metadata that can be associated with a type of a SD-JWT VC, as well as a method for retrieving the Type Metadata and processing rules. This Type Metadata is intended to be used, among other things, for the following purposes:
 
- * Developers of Issuers and Verifiers can use the Typ Metadata to understand the
+ * Developers of Issuers and Verifiers can use the Type Metadata to understand the
    semantics of the type and the associated rules. While in some cases,
-   Issuers are the parties that define types (credential formats), this is
+   Issuers are the parties that define types, this is
    not always the case. For example, a type can be defined by a
    standardization body or a community.
- * Verifiers can use the metadata to determine whether a credential is valid
+ * Verifiers can use the Type Metadata to determine whether a credential is valid
    according to the rules of the type. For example, a Verifier can check
    whether a credential contains all required claims and whether the claims
    are selectively disclosable.
- * Wallets can use the metadata to display the credential in a way that is
-   consistent with the Issuer's intent.
 
-Applications using Type Metadata defined in this specification are called "Consumers" in the following. This typically includes Issuers, Verifiers, and Wallets.
+Type Metadata can be retrieved as described in (#retrieving-type-metadata).
 
 ## Type Metadata Example {#type-metadata-example}
 
 All examples in this section are non-normative.
 
-Type Metadata about the value of a `vct` claim value can be retrieved as described in (#retrieving-type-metadata).
-
 ```json
 {
   "vct": "https://betelgeuse.example.com/education_credential",
```

Comment:
Is the `vct` field required in a metadata JSON object? This would break compatibility with content addressable storage systems. For example, if I want to use `ipfs://<hash>` as a `vct` uri. It would be easy to use the ipfs-uri in the JWT, but it's impossible to put it inside of the metadata JSON object, since it would require knowing the hash of that object before calculating it.

Seems like `vct` needs to be optional here.

Reply:
Yes, I agree that this would not work with content addressable storage systems the way it is defined now. It is a chicken and egg problem. IMO, we should support content addressable hash URIs for `vct` and make `vct` optional in the type metadata.

@danielfett @bc-pi is there is a security-related reason to make `vct` in type metadata mandatory?

Reply:
As it is designed at the moment, vct is at least required for usage in glue documents. We need to figure out how to address this.

Reply:
I'm not sure that storage and retrieval of Type Metadata from content addressable storage systems is or should a goal.

Reply:
IMO, there is a community that uses this and SD-JWT VC and if possible we should help them to support this.

Reply:
> I'm not sure that storage and retrieval of Type Metadata from content addressable storage systems is or should a goal.

This makes me sad. If sd-jwt vc actively is trying to disallow usage by CAS systems we won't be able to adopt it. 
There are a few reasons why I think using CAS (over something like http) is nice:
1. Less trust assumptions, if the url is the hash, I know what I get is correct no matter who gives it to me
2. Less technical assumptions, hashes are generally technically simpler than https signatures
3. Content can't be arbitrarily changed, with https the DNS controller ultimately decides what content is served

Reply:
The goal should be to not prohibit it, we don't need to add normative language around that.

Reply:
As this question is not directly related to the PR, please continue the discussion in this issue: https://github.com/oauth-wg/oauth-sd-jwt-vc/issues/211

Reply:
@danielfett I would say it's actually directly related ro the PR since the issue only exist in the context of the `vct` field inside of the type metadata object.